server/battles/BattleResultProcessor.cpp

```diff
@@ -188,15 +188,15 @@ FinishingBattleHelper::FinishingBattleHelper(const CBattleInfoCallback & info, c
 {
 	if (result.winner == BattleSide::ATTACKER)
 	{
-		winnerHero = info.getBattle()->getSideHero(BattleSide::ATTACKER);
-		loserHero = info.getBattle()->getSideHero(BattleSide::DEFENDER);
+		winnerId = info.getBattle()->getSideArmy(BattleSide::ATTACKER)->id;
+		loserId = info.getBattle()->getSideArmy(BattleSide::DEFENDER)->id;
```

Comment:
I think there might be cases where winner army != winner hero - like town sieges. Or not?